core/src/tasks/group/updateCalculatedGroups.ts

```diff
@@ -21,21 +21,37 @@ export class GroupsUpdateCalculatedGroups extends CLSTask {
       "groups-calculation-delay-minutes"
     );
     const delayMinutes = parseInt(setting.value);
-    const lastCheckTime = Moment().subtract(delayMinutes, "minutes");
+    const lastCheckTime = Moment().subtract(delayMinutes, "minutes").toDate();
 
+    const groupsToRun: Group[] = [];
     const calculatedGroups = await Group.scope(null).findAll({
-      where: {
-        type: "calculated",
-        state: "ready",
-        calculatedAt: {
-          [Op.or]: {
-            [Op.eq]: null,
-            [Op.lt]: lastCheckTime,
-          },
-        },
-      },
+      where: { type: "calculated" },
     });
 
-    await Promise.all(calculatedGroups.map((group) => group.run()));
+    for (const group of calculatedGroups) {
+      if (group.state === "ready" && !group.calculatedAt) {
```

Comment:
acceeptance testing this. I think there are some more cases, especially on first time running.

For example, if the state is "updating" and `!group.calculatedAt` is not handled.

Reply:
since most of these assume the `group.calculatedAt`, maybe deal with that first
```
const lastRunAt = group.calculatedAt;
if(!lastRunAt) {
  // handle ready and updating cases as relevant.
}
else if (lastRunAt < lastCheckTime.getTime()) {
  // it's been awhile. handle based on state
}
```